Lib/tarfile.py

```diff
@@ -1969,11 +1969,13 @@ def add(self, name, arcname=None, recursive=True, *, filter=None):
                 self.addfile(tarinfo, f)
 
         elif tarinfo.isdir():
-            self.addfile(tarinfo)
+            if tarinfo.name != '.':
```

Comment:
I'm not deeply familiar enough with tar to know the answer to this, but should this be hard-coded to `'.'` or use `os.curdir`? (here and below as well)

Reply:
I have made the requested changes; please review again